anvio/drivers/prodigal.py

```diff
@@ -94,78 +99,56 @@ def check_version(self):
-        amino_acid_sequences_dict = {}
 
+        # Set up the output files.
+        log_file_path = os.path.join(output_dir, '00_log.txt')
         self.genes_in_contigs = os.path.join(output_dir, 'contigs.genes')
         self.amino_acid_sequences_in_contigs = os.path.join(output_dir, 'contigs.amino_acid_sequences')
 
-        log_file_path = os.path.join(output_dir, '00_log.txt')
-
+        # Put some nice logging info.
         self.run.warning('', header='Finding ORFs in contigs', lc='green')
         self.run.info('Genes', self.genes_in_contigs)
         self.run.info('Amino acid sequences', self.amino_acid_sequences_in_contigs)
         self.run.info('Log file', log_file_path)
 
-        cmd_line = ['prodigal', '-i', fasta_file_path, '-o', self.genes_in_contigs, '-a', self.amino_acid_sequences_in_contigs]
-
-        if self.prodigal_translation_table:
-            cmd_line.extend(['-g', self.prodigal_translation_table])
-            self.run.warning("Prodigal translation table is set to '%s' (whatever you did has worked so far, but "
-                             "keep an eye for errors from prodigal in case it doesn't like your translation table "
-                             "parameter). This means we will not use prodigal in the metagenomics mode, due to this "
-                             "issue: https://github.com/hyattpd/Prodigal/issues/19. If that issue is closed, and you "
-                             "are reading this message, then please contact an anvi'o developer." % str(self.prodigal_translation_table))
-        else:
-            cmd_line.extend(['-p', 'meta'])
-
         self.run.warning("Anvi'o will use 'prodigal' by Hyatt et al (doi:10.1186/1471-2105-11-119) to identify open "
-                         "reading frames in your data. When you publish your findings, please do not forget to properly "
-                         "credit their work.", lc='green', header="CITATION")
+                         "reading frames in your data. When you publish your findings, please do not forget to "
+                         "properly credit their work.", lc='green', header="CITATION")
 
         self.progress.new('Processing')
-        self.progress.update('Identifying ORFs in contigs ...')
-
-        utils.run_command(cmd_line, log_file_path)
-
-        if not os.path.exists(self.amino_acid_sequences_in_contigs):
-            self.progress.end()
-            raise ConfigError("Something went wrong with prodigal, and it failed to generate the "
-                              "expected output :/ Fortunately, this log file should tell you what "
-                              "might be the problem: '%s'. Please do not forget to include this "
-                              "file if you were to ask for help." % log_file_path)
+        self.progress.update('Identifying ORFs using %s' % (('%d threads.' % self.num_threads) if self.num_threads > 1 else 'a single thread (boo).'))
```

Comment:
I wonder if there is an easy way to show the number of active threads. Because some threads will be done with their files quicker than others.

Reply:
The number of active threads is available in `threading.enumerate()`.  This will include the calling thread and any other threads that may have been spawned.  First thought I had was just to check for `AnviThread`s as that is what this class will spawn.  But it's conceivable that something else could spawn an `AnviThread`.  Then I thought, just give the threading ops threads specific names and search against that.  Meh, kind of annoying to do that.

Then I thought of something like this:

```python
        living_worker_threads = [thread for thread in self.threads if thread.is_alive()]
        while len(living_worker_threads) > 0:
            if self.logger:
                self.logger.run.info('Info', f'living_worker_threads: {len(living_worker_threads)}')

            time.sleep(1)
            living_worker_threads = [thread for thread in self.threads if thread.is_alive()]
```

This would basically be a hacky version of calling `join()` on all threads in `self.threads` a `for` loop, except that it will print out info about how many threads are currently running.  On the other hand maybe it's kind of annoying to print a message like that every second...what do you think?

(Of course the `[thread for thread in self.threads if thread.is_alive()]` bit could become a class method so it's easier to call.)